### DISCUSSION TITLE
Nextjs Cache, vercel and local interaction

### DISCUSSION THREAD

**Discussion by DoIevenLift** [vercel/next.js#63518] [Help]
### Summary

Hey all, 

Apologies for a potentially noob question / discussion topic.

I am encountering a weird kind of 'issue' with the cache that has only started happening in the last few days. For reference I am using nextjs version 14.0.4 & supabase-js.

The cache 'issue' is only showing on a single page (or collection of pages) and what is happening is that the GET and POST request is being memorized, resulting in a stale value being used in proceeding logic. From the Nextjs docs - https://nextjs.org/docs/app/building-your-application/caching#request-memoization - the POST request shouldn't be memorized.

These are the requests inside the page /locations/[locationId]/service/[serviceId]:

![image](https://github.com/vercel/next.js/assets/78464058/1c6f7015-68da-4af6-9584-16430b342660)

As you can see there are a number of requests where the cache gets hit, one of those being a POST. 

From the Vercel docs - https://vercel.com/docs/deployments/troubleshoot-a-build#understanding-build-cache it looks like everything under .next/cache/** is carried across builds? 

Further to the nextjs docs it looks like what could be happening is a full route cache whereby all the values from the requests etc are being cached and then returned - which seems a bit strange to me, given Supabase-js doesn't use an extended 'fetch' and if I look in the vercel dashboard at the cache I can see that this is never hit:

![image](https://github.com/vercel/next.js/assets/78464058/26eb2d0c-ea8a-4b94-85a6-02b5f68bcffd)

So my questions / discussions are: 
1) How do you determine what pages / routes / fetches of the application are stored inside .next/cache ? 
2) On a full route, how do you test or identify that the it is actually being cached and if you don't want it where do you apply the 'dynamic = 'force-dynamic' or 'revalidate = 0' if the requests are handled inside a server component where the data is fetched via supabase-js inside server actions? 
3) Are there methods to 'remove' a cache across builds without manually pressing 'redeploy' or using an env var for Vercel ?

Appreciate any pointers / references

Thanks!

**Comment by RadovanPelka** [comment]
@DoIevenLift 
Check this https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config#dynamic

this helped me:

```
export const dynamic = "force-dynamic";
export const revalidate = 0;
```

**Reply by DoIevenLift** [reply]
Hey @RadovanPelka , 

Thanks for the link and reply. Have added those in at layout level and will keep testing. 

Just for my knowledge moving forward, is this what 'could' be happening where everything under the 'layout.tsx' is being fully cached / is that a possibility? 

Thanks again for your time

**Reply by DoIevenLift** [reply]
Just wanted to give an update after testing + rebuilding (not redeploying to Vercel). 

I've narrowed this down to what I think is 98% certain a cache issue.  For some more context the value is retrieved by hitting the db through a GET call. 

If I look at the Supabase API logs this GET call doesn't even 'fire':

![image](https://github.com/vercel/next.js/assets/78464058/8b1beae7-40bb-4047-b405-7bf9ff39c904)

If I look at the Nextjs logs (locally) - this matches:

![image](https://github.com/vercel/next.js/assets/78464058/f134db86-ec35-4338-9597-bca7ff1c00d0)

Things that have been done:
- Added force dynamic and revalidation = 0.
- Rebuild the function that does the GET Call
- Deleted .next folder
- Did yarn build

So if we compare to the original screenshot of the Nextjs logs we are down a few cache hits (progress) but it still doesn't answer why/how this GET call, which is using supabase-js (which shouldn't cache because doesn't use the configured 'fetch') is being cached and what could be causing it to retrieve a stale value. 

Thanks again
